src/main/java/org/terasology/fluid/system/FluidAuthoritySystem.java

```diff
@@ -100,9 +101,9 @@ public void fillFluidContainerItem(ActivateEvent event, EntityRef item, FluidCon
                 final EntityRef removedItem = inventoryManager.removeItem(owner, event.getInstigator(), item, false, 1);
                 //TODO: replace with better fluid handling, maybe by new CoreFluids module
                 if (removedItem != null && block.isWater()) {
-                    // Set the contents of this fluid container and fill it up to max capacity.
+                    // Set the contents of this fluid container and fill it up to (current volume + filling amount).
                     FluidUtils.setFluidForContainerItem(removedItem, "Fluid:Water",
-                            removedItem.getComponent(FluidContainerItemComponent.class).maxVolume);
+                            removedItem.getComponent(FluidContainerItemComponent.class).volume + removedItem.getComponent(FluidContainerItemComponent.class).fillingAmount);
```

Comment:
As you are using the `FluidContainerItemComponent` twice here it would be beneficial to retrieve the component once before, and use that local variable instead.